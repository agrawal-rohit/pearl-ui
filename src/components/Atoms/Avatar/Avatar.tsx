import React from "react";
import { useMolecularComponentConfig } from "../../../hooks/useMolecularComponentConfig";
import Box from "../../Atoms/Box/Box";
import { ImageProps } from "../../Molecules/Image/Image";

type AvatarProps = ImageProps & {
  /** The size of the avatar */
  size?: string;
  /** The variant of the avatar */
  variant?: string;
};

// TODO: Add main user image render
// TODO: Add icon fallback based on user name
// TODO: Add support for custom fallback icon
// TODO: Add icon badge to avatar

/** The Avatar component is used to represent a user, and displays the profile picture, initials or fallback icon. */
const Avatar: React.FC<AvatarProps> = ({ children, ...rest }) => {
  const props = useMolecularComponentConfig("Avatar", rest, {
    size: rest.size,
    variant: rest.variant,
  });

  return <Box></Box>;
};

export default Avatar;
