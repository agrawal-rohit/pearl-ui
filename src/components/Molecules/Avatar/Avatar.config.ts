import { MolecularComponentConfig } from "./../../../theme/src/types";
export default {
  parts: ["root", "text"],
  baseStyle: {
    root: {
      borderRadius: "full",
    },
  },
  sizes: {
    s: {
      root: {
        width: 40,
        height: 40,
      },
      text: {
        variant: "caption",
      },
    },
    m: {
      root: {
        width: 55,
        height: 55,
      },
      text: {
        variant: "p1",
        fontSize: "m",
        lineHeight: "3xl",
      },
    },
    l: {
      root: {
        width: 70,
        height: 70,
      },
      text: {
        variant: "p1",
        fontSize: "m",
        lineHeight: "3xl",
      },
    },
    xl: {
      root: {
        width: 85,
        height: 85,
      },
      text: {
        variant: "p1",
        fontSize: "4xl",
        lineHeight: "8xl",
      },
    },
  },
  defaults: {
    size: "m",
  },
} as MolecularComponentConfig;
