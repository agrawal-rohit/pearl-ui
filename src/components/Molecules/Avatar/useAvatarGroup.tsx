import { createContext, useContext } from "react";
import {
  ComponentSizes,
  ComponentVariants,
  ResponsiveValue,
} from "../../../theme/src/types";

interface IAvatarGroupContext {
  /** Size of all the children avatars in the group. */
  size?: ResponsiveValue<ComponentSizes<"Avatar">>;
  /** Variant of all the children avatars in the group. */
  variant?: ResponsiveValue<ComponentVariants<"Avatar">>;
}

export const avatarGroupContext = createContext({} as IAvatarGroupContext);

/**
 * Hook to get access to the state of a Checkbox group
 */
export const useAvatarGroup = () =>
  useContext(avatarGroupContext) as IAvatarGroupContext;
