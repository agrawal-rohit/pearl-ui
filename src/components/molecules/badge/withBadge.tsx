import React, { useCallback, useMemo, useState } from "react";
import { LayoutChangeEvent, Platform } from "react-native";
import Box from "../../atoms/box/box";
import Badge, { BadgeProps } from "./badge";

export interface WithBadgeOptions {
  /**
   * The position of the badge with respect to the provided base component
   *
   * @default "topRight"
   */
  placement?: "topLeft" | "topRight" | "bottomLeft" | "bottomRight";
  /**
   * Amount of extra spacing to add between the badge and the base component
   *
   * @default 5
   */
  offset?: number;
  /**
   * Whether the badge is visible or not
   *
   * @default false
   */
  hidden?: boolean;
}

/**
 * A higher-order component which can be used to add a badge to any target component.
 * @param badgeValue The value to be displayed inside the badge
 * @param options A set of options for customizing the look of the badge
 * @returns
 */
const withBadge =
  <P extends {}>(
    badgeValue: string | number | React.ReactElement | undefined = undefined,
    options: WithBadgeOptions & BadgeProps = {}
  ) =>
  (WrappedComponent: React.FC<P>) =>
  (props: P) => {
    // Destructure options with default values
    const {
      placement = "topRight",
      offset = 5,
      hidden = false,
      ...badgeProps
    } = options;

    // State variables to hold the width of the base component and the badge
    const [baseComponentWidth, setBaseComponentWidth] = useState(0);
    const [badgeWidth, setBadgeWidth] = useState(0);

    // Function to compute the position of the badge based on the placement option
    const computePositionForBadge = useMemo(() => {
      const positionValue = -1 * offset;

      if (placement === "topLeft")
        return { top: positionValue, left: positionValue };

      if (placement === "topRight")
        return { top: positionValue, right: positionValue };

      if (placement === "bottomLeft")
        return { bottom: positionValue, left: positionValue };

      // Return bottom right position by default
      return { bottom: positionValue, right: positionValue };
    }, [placement, offset]);

    // Function to update the width of the badge when its layout changes
    const onBadgeLayoutChange = (event: LayoutChangeEvent) => {
      const { width } = event.nativeEvent.layout;
      setBadgeWidth(width);
    };

    // Function to update the width of the base component when its layout changes
    const onBaseComponentLayoutChange = (event: LayoutChangeEvent) => {
      const { width } = event.nativeEvent.layout;
      setBaseComponentWidth(width);
    };

    // Render the wrapped component with the badge
    return (
      <Box>
        <Box alignSelf="flex-start">
          <WrappedComponent
            {...(props as P)}
            onLayout={onBaseComponentLayoutChange}
          />

          {!hidden && (
            <Badge
              {...badgeProps}
              onLayout={onBadgeLayoutChange}
              position="absolute"
              zIndex="overlay"
              style={computePositionForBadge}
            >
              {badgeValue}
            </Badge>
          )}
        </Box>
      </Box>
    ) as React.ReactElement<P>;
  };

export default withBadge;
