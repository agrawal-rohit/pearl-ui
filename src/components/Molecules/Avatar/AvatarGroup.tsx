import React, { ReactElement } from "react";
import {
  ComponentSizes,
  ComponentVariants,
  FinalPearlTheme,
  PaletteColors,
  ResponsiveValue,
} from "../../../theme/src/types";
import { spacingStyleFunction } from "../../../theme/src/styleFunctions";
import Box, { BoxProps } from "../../Atoms/Box/Box";
import { ZStack } from "../../Atoms/Stack/Stack";
import Avatar from "./Avatar";
import { avatarGroupContext } from "./useAvatarGroup";
import { useStyleProps } from "../../../hooks/useStyleProps";

export type AvatarGroupProps = BoxProps & {
  /** Size of all the children avatars in the group. */
  size?: ResponsiveValue<ComponentSizes<"Avatar">>;
  /** Variant of all the children avatars in the group. */
  variant?: ResponsiveValue<ComponentVariants<"Avatar">>;
  /** The spacing between the avatars */
  spacing?: ResponsiveValue<keyof FinalPearlTheme["spacing"]>;
  /** The background color of the circle which shows the "+X" label of remaining avatars */
  truncatedBackgroundColor?: ResponsiveValue<PaletteColors>;
  /** A custom component to show in place of the truncated avatars  */
  customTruncatedComponent?: React.ReactElement;
  /** Maximum number of avatars to show. It'll truncate the avatars and show a "+X" label (where X is the remaining avatars) */
  max?: number;
};

const AvatarGroup: React.FC<AvatarGroupProps> = ({
  children,
  max = 3,
  ...rest
}) => {
  const convertedElementSpacing = useStyleProps({ marginLeft: rest.spacing }, [
    ...spacingStyleFunction,
  ]);
  const avatarChildren = React.Children.toArray(children);

  const renderAvatars = () => {
    return React.Children.map(avatarChildren, (child, index) => {
      const shouldBreakLoop = index > max;
      const shouldRenderAvatar = index + 1 <= max;

      if (shouldBreakLoop) return;

      if (shouldRenderAvatar)
        return React.cloneElement(child as ReactElement, {
          ...(child as ReactElement).props,
          style: {
            ...(child as ReactElement).props.style,
            marginLeft: index * convertedElementSpacing.style.marginLeft,
          },
        });
      else {
        if (rest.customTruncatedComponent)
          return React.cloneElement(rest.customTruncatedComponent, {
            ...rest.customTruncatedComponent.props,
            remainingAvatars: avatarChildren.length - max,
            style: {
              ...rest.customTruncatedComponent.props.style,
              marginLeft: index * convertedElementSpacing.style.marginLeft,
            },
          });

        return (
          <Avatar
            name={`+${avatarChildren.length - max}`}
            getInitials={(name: string) => name}
            backgroundColor={rest.truncatedBackgroundColor}
            style={{
              marginLeft: index * convertedElementSpacing.style.marginLeft,
            }}
          ></Avatar>
        );
      }
    });
  };

  return (
    <Box {...rest}>
      <avatarGroupContext.Provider
        value={{
          size: rest.size,
          variant: rest.variant,
        }}
      >
        <ZStack reversed>{renderAvatars()}</ZStack>
      </avatarGroupContext.Provider>
    </Box>
  );
};

export default AvatarGroup;
