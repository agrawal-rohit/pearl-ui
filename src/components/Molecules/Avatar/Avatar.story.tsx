import React from "react";
import { storiesOf } from "@storybook/react-native";
import Screen from "../../Atoms/Screen/Screen";
import Avatar, { AvatarProps } from "./Avatar";
import Stack from "../../Atoms/Stack/Stack";
import Icon from "../../Atoms/Icon/Icon";
import withBadge from "../Badge/withBadge";

storiesOf("Avatar", module)
  .addDecorator((getStory) => <Screen>{getStory()}</Screen>)
  .add("Sizes", () => (
    <>
      <Stack direction="horizontal" spacing="s">
        <Avatar
          src="https://pbs.twimg.com/profile_images/1419369145058041856/eshLFaDy_400x400.jpg"
          size="s"
        />
        <Avatar
          src="https://pbs.twimg.com/profile_images/1419369145058041856/eshLFaDy_400x400.jpg"
          size="m"
        />
        <Avatar
          src="https://pbs.twimg.com/profile_images/1419369145058041856/eshLFaDy_400x400.jpg"
          size="l"
        />
        <Avatar
          src="https://pbs.twimg.com/profile_images/1419369145058041856/eshLFaDy_400x400.jpg"
          size="xl"
        />
      </Stack>

      <Stack direction="horizontal" spacing="s" mt="s">
        <Avatar name="Rohit Agrawal" size="s" />
        <Avatar name="John" size="m" />
        <Avatar name="Some guy" size="l" />
        <Avatar name="YOYO" size="xl" />
      </Stack>
    </>
  ))
  .add("Fallbacks", () => (
    <Stack direction="horizontal" spacing="s">
      <Avatar name="Rohit Agrawal" backgroundColor="cyan" />
      <Avatar
        backgroundColor="neutral.200"
        fallbackComponent={
          <Icon
            iconFamily="FontAwesome"
            iconName="user-circle"
            color="neutral.600"
            rawSize={55}
          />
        }
      />
    </Stack>
  ))
  .add("With Badge", () => {
    const OnlineAvatar = withBadge<AvatarProps>(undefined, {
      placement: "bottomRight",
      backgroundColor: "success.500",
      size: "s",
      minW: 15,
      h: 15,
      offset: 0,
    })(Avatar);

    const OfflineAvatar = withBadge<AvatarProps>(0, {
      placement: "bottomRight",
      backgroundColor: "danger.500",
      size: "s",
      minW: 15,
      h: 15,
      offset: 0,
    })(Avatar);

    const AvatarWithEdit = withBadge<AvatarProps>(
      <Icon iconFamily="Ionicons" iconName="pencil" size="s" color="white" />,
      {
        placement: "bottomRight",
        size: "m",
      }
    )(Avatar);

    return (
      <Stack direction="horizontal" spacing="s">
        <OnlineAvatar src="https://pbs.twimg.com/profile_images/1419369145058041856/eshLFaDy_400x400.jpg" />

        <OfflineAvatar name="Rohit Agrawal" />

        <AvatarWithEdit />
      </Stack>
    );
  });
