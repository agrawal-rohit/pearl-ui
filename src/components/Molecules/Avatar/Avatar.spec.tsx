import React from "react";
import { render } from "@testing-library/react-native";
import Avatar, { AvatarProps } from "./avatar";
import { ThemeProvider } from "../../../theme/src/theme-context";
import Icon, { IconProps } from "../../atoms/icon/icon";
import withBadge from "../badge/withBadge";
import Stack from "../../atoms/stack/stack";
import AvatarGroup from "./avatar-group";

jest.useFakeTimers();

describe("Molecules/Avatar", () => {
  it("passes the snapshot test for different sizes", () => {
    const tree = render(
      <ThemeProvider>
        <>
          <Stack direction="horizontal" spacing="s">
            <Avatar
              src="https://pbs.twimg.com/profile_images/1419369145058041856/eshLFaDy_400x400.jpg"
              size="s"
            />
            <Avatar
              src="https://pbs.twimg.com/profile_images/1419369145058041856/eshLFaDy_400x400.jpg"
              size="m"
            />
            <Avatar
              src="https://pbs.twimg.com/profile_images/1419369145058041856/eshLFaDy_400x400.jpg"
              size="l"
            />
            <Avatar
              src="https://pbs.twimg.com/profile_images/1419369145058041856/eshLFaDy_400x400.jpg"
              size="xl"
            />
          </Stack>

          <Stack direction="horizontal" spacing="s" mt="s">
            <Avatar backgroundColor="red" name="Rohit Agrawal" size="s" />
            <Avatar backgroundColor="red" name="John" size="m" />
            <Avatar backgroundColor="red" name="Some guy" size="l" />
            <Avatar backgroundColor="red" name="YOYO" size="xl" />
          </Stack>
        </>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test for fallback support", () => {
    const tree = render(
      <ThemeProvider>
        <Stack direction="horizontal" spacing="s">
          <Avatar backgroundColor="red" name="Rohit Agrawal" />
          <Avatar
            src="https://4kwallpapers.com/imas/wallpapers/macos-big-sur-apple-layers-fluidic-colorful-wwdc-stock-2560x1440-1455.jpg"
            backgroundColor="red"
            fallbackSource={{
              uri: "https://cdn.segmentnext.com/wp-content/themes/segmentnext/images/no-image-available.jpg",
            }}
          />
          <Avatar
            backgroundColor="neutral.200"
            fallbackComponent={
              <Icon
                iconFamily="FontAwesome"
                iconName="user-circle"
                color="neutral.600"
                rawSize={40}
              />
            }
          />
        </Stack>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("withBadge HOC works as expected", () => {
    const Wrapper = () => {
      const OnlineAvatar = withBadge<AvatarProps>(undefined, {
        placement: "bottomRight",
        backgroundColor: "success.500",
        size: "s",
        minW: 15,
        h: 15,
        offset: 0,
      })(Avatar);

      const OfflineAvatar = withBadge<AvatarProps>(0, {
        placement: "bottomRight",
        backgroundColor: "danger.500",
        size: "s",
        offset: 0,
      })(Avatar);

      const AvatarWithEdit = withBadge<AvatarProps>(
        <Icon iconFamily="Ionicons" iconName="pencil" size="s" color="white" />,
        {
          placement: "bottomRight",
          size: "m",
          onPress: () => {
            console.log("PRESSED!");
          },
        }
      )(Avatar);
      return (
        <Stack direction="horizontal" spacing="s">
          <OnlineAvatar
            backgroundColor="cyan"
            src="https://pbs.twimg.com/profile_images/1419369145058041856/eshLFaDy_400x400.jpg"
          />

          <OfflineAvatar backgroundColor="cyan" name="Rohit Agrawal" />

          <AvatarWithEdit backgroundColor="red" />
        </Stack>
      );
    };

    const root = render(
      <ThemeProvider>
        <Wrapper />
      </ThemeProvider>
    );

    const tree = root.toJSON();

    expect(tree).toMatchSnapshot();
    expect(root.getByText("0")).toBeTruthy();
  });

  it("passes the snapshot test when used in a group", () => {
    const tree = render(
      <ThemeProvider>
        <AvatarGroup
          spacing="2xl"
          max={2}
          truncatedBackgroundColor="neutral.200"
        >
          <Avatar
            name="Rohit Agrawal"
            src="https://pbs.twimg.com/profile_images/1419369145058041856/eshLFaDy_400x400.jpg"
          />
          <Avatar
            name="Rohit Agrawal"
            src="https://avatars.githubusercontent.com/u/29514438?s=400&u=d194d5de8df93f55038130ccc66429f94f8f900f&v=4"
          />
          <Avatar
            name="Rohit Agrawal"
            src="https://instagram.fdel1-3.fna.fbcdn.net/v/t51.2885-19/s320x320/160616189_1075891466264003_198594308312142696_n.jpg?_nc_ht=instagram.fdel1-3.fna.fbcdn.net&_nc_cat=111&_nc_ohc=-URhWepekUsAX_wt_-J&edm=ABfd0MgBAAAA&ccb=7-4&oh=00_AT-caDqy7XqUUTkGv5_QytlFTxtdZ2wVhZDgB4vU3Jl2qQ&oe=61DF5972&_nc_sid=7bff83"
          />
        </AvatarGroup>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
