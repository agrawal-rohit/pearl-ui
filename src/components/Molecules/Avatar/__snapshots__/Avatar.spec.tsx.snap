// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/Avatar passes the snapshot test for different sizes 1`] = `
Array [
  <View
    direction="horizontal"
    onLayout={[Function]}
    spacing="s"
    style={
      Object {
        "alignSelf": "flex-start",
        "borderRadius": undefined,
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "borderRadius": undefined,
          "display": "flex",
          "flexDirection": "row",
          "marginRight": 12,
        }
      }
    >
      <View
        accessibilityRole="image"
        accessible={true}
        isCached={false}
        loaderType="spinner"
        onLayout={[Function]}
        size="s"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F9FC",
            "borderBottomLeftRadius": undefined,
            "borderBottomRightRadius": undefined,
            "borderRadius": 1000,
            "borderTopLeftRadius": undefined,
            "borderTopRightRadius": undefined,
            "display": "flex",
            "height": 40,
            "justifyContent": "center",
            "width": 40,
          }
        }
        tint="dark"
        transitionDuration={300}
      >
        <Image
          backgroundColor="neutral.100"
          borderRadius="l"
          isCached={false}
          loaderType="spinner"
          size="s"
          source={
            Object {
              "uri": "https://pbs.twimg.com/profile_images/1419369145058041856/eshLFaDy_400x400.jpg",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderBottomLeftRadius": undefined,
                "borderBottomRightRadius": undefined,
                "borderRadius": 1000,
                "borderTopLeftRadius": undefined,
                "borderTopRightRadius": undefined,
                "height": "100%",
                "width": "100%",
                "zIndex": 2,
              },
            ]
          }
          tint="dark"
          transitionDuration={300}
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "borderRadius": undefined,
          "display": "flex",
          "flexDirection": "row",
          "marginRight": 12,
        }
      }
    >
      <View
        accessibilityRole="image"
        accessible={true}
        isCached={false}
        loaderType="spinner"
        onLayout={[Function]}
        size="m"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F9FC",
            "borderBottomLeftRadius": undefined,
            "borderBottomRightRadius": undefined,
            "borderRadius": 1000,
            "borderTopLeftRadius": undefined,
            "borderTopRightRadius": undefined,
            "display": "flex",
            "height": 55,
            "justifyContent": "center",
            "width": 55,
          }
        }
        tint="dark"
        transitionDuration={300}
      >
        <Image
          backgroundColor="neutral.100"
          borderRadius="l"
          isCached={false}
          loaderType="spinner"
          size="m"
          source={
            Object {
              "uri": "https://pbs.twimg.com/profile_images/1419369145058041856/eshLFaDy_400x400.jpg",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderBottomLeftRadius": undefined,
                "borderBottomRightRadius": undefined,
                "borderRadius": 1000,
                "borderTopLeftRadius": undefined,
                "borderTopRightRadius": undefined,
                "height": "100%",
                "width": "100%",
                "zIndex": 2,
              },
            ]
          }
          tint="dark"
          transitionDuration={300}
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "borderRadius": undefined,
          "display": "flex",
          "flexDirection": "row",
          "marginRight": 12,
        }
      }
    >
      <View
        accessibilityRole="image"
        accessible={true}
        isCached={false}
        loaderType="spinner"
        onLayout={[Function]}
        size="l"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F9FC",
            "borderBottomLeftRadius": undefined,
            "borderBottomRightRadius": undefined,
            "borderRadius": 1000,
            "borderTopLeftRadius": undefined,
            "borderTopRightRadius": undefined,
            "display": "flex",
            "height": 70,
            "justifyContent": "center",
            "width": 70,
          }
        }
        tint="dark"
        transitionDuration={300}
      >
        <Image
          backgroundColor="neutral.100"
          borderRadius="l"
          isCached={false}
          loaderType="spinner"
          size="l"
          source={
            Object {
              "uri": "https://pbs.twimg.com/profile_images/1419369145058041856/eshLFaDy_400x400.jpg",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderBottomLeftRadius": undefined,
                "borderBottomRightRadius": undefined,
                "borderRadius": 1000,
                "borderTopLeftRadius": undefined,
                "borderTopRightRadius": undefined,
                "height": "100%",
                "width": "100%",
                "zIndex": 2,
              },
            ]
          }
          tint="dark"
          transitionDuration={300}
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "borderRadius": undefined,
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityRole="image"
        accessible={true}
        isCached={false}
        loaderType="spinner"
        onLayout={[Function]}
        size="xl"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F9FC",
            "borderBottomLeftRadius": undefined,
            "borderBottomRightRadius": undefined,
            "borderRadius": 1000,
            "borderTopLeftRadius": undefined,
            "borderTopRightRadius": undefined,
            "display": "flex",
            "height": 85,
            "justifyContent": "center",
            "width": 85,
          }
        }
        tint="dark"
        transitionDuration={300}
      >
        <Image
          backgroundColor="neutral.100"
          borderRadius="l"
          isCached={false}
          loaderType="spinner"
          size="xl"
          source={
            Object {
              "uri": "https://pbs.twimg.com/profile_images/1419369145058041856/eshLFaDy_400x400.jpg",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderBottomLeftRadius": undefined,
                "borderBottomRightRadius": undefined,
                "borderRadius": 1000,
                "borderTopLeftRadius": undefined,
                "borderTopRightRadius": undefined,
                "height": "100%",
                "width": "100%",
                "zIndex": 2,
              },
            ]
          }
          tint="dark"
          transitionDuration={300}
        />
      </View>
    </View>
  </View>,
  <View
    direction="horizontal"
    onLayout={[Function]}
    spacing="s"
    style={
      Object {
        "alignSelf": "flex-start",
        "borderRadius": undefined,
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginTop": 12,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "borderRadius": undefined,
          "display": "flex",
          "flexDirection": "row",
          "marginRight": 12,
        }
      }
    >
      <View
        onLayout={[Function]}
        size="s"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ff0000",
            "borderRadius": 1000,
            "display": "flex",
            "height": 40,
            "justifyContent": "center",
            "width": 40,
          }
        }
      >
        <Text
          accessibilityRole="text"
          accessible={true}
          allowFontScaling={true}
          scaleFontSize={true}
          style={
            Object {
              "color": "#FFFFFF",
              "display": "flex",
              "fontFamily": "Poppins-Regular",
              "fontSize": 13.799999999999999,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 20,
            }
          }
          variant="caption"
        >
          RA
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "borderRadius": undefined,
          "display": "flex",
          "flexDirection": "row",
          "marginRight": 12,
        }
      }
    >
      <View
        onLayout={[Function]}
        size="m"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ff0000",
            "borderRadius": 1000,
            "display": "flex",
            "height": 55,
            "justifyContent": "center",
            "width": 55,
          }
        }
      >
        <Text
          accessibilityRole="text"
          accessible={true}
          allowFontScaling={true}
          scaleFontSize={true}
          style={
            Object {
              "color": "#FFFFFF",
              "display": "flex",
              "fontFamily": "Poppins-Regular",
              "fontSize": 18.4,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 28,
            }
          }
          variant="p1"
        >
          J
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "borderRadius": undefined,
          "display": "flex",
          "flexDirection": "row",
          "marginRight": 12,
        }
      }
    >
      <View
        onLayout={[Function]}
        size="l"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ff0000",
            "borderRadius": 1000,
            "display": "flex",
            "height": 70,
            "justifyContent": "center",
            "width": 70,
          }
        }
      >
        <Text
          accessibilityRole="text"
          accessible={true}
          allowFontScaling={true}
          scaleFontSize={true}
          style={
            Object {
              "color": "#FFFFFF",
              "display": "flex",
              "fontFamily": "Poppins-Regular",
              "fontSize": 18.4,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 28,
            }
          }
          variant="p1"
        >
          SG
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "borderRadius": undefined,
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <View
        onLayout={[Function]}
        size="xl"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ff0000",
            "borderRadius": 1000,
            "display": "flex",
            "height": 85,
            "justifyContent": "center",
            "width": 85,
          }
        }
      >
        <Text
          accessibilityRole="text"
          accessible={true}
          allowFontScaling={true}
          scaleFontSize={true}
          style={
            Object {
              "color": "#FFFFFF",
              "display": "flex",
              "fontFamily": "Poppins-Regular",
              "fontSize": 29.9,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 38,
            }
          }
          variant="p1"
        >
          Y
        </Text>
      </View>
    </View>
  </View>,
]
`;

exports[`Molecules/Avatar withBadge HOC works as expected 1`] = `
<View
  direction="horizontal"
  onLayout={[Function]}
  spacing="s"
  style={
    Object {
      "alignSelf": "flex-start",
      "borderRadius": undefined,
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "borderRadius": undefined,
        "display": "flex",
        "flexDirection": "row",
        "marginRight": 12,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignSelf": "flex-start",
          "borderRadius": undefined,
          "display": "flex",
        }
      }
    >
      <View
        accessibilityRole="image"
        accessible={true}
        isCached={false}
        loaderType="spinner"
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F9FC",
            "borderBottomLeftRadius": undefined,
            "borderBottomRightRadius": undefined,
            "borderRadius": 1000,
            "borderTopLeftRadius": undefined,
            "borderTopRightRadius": undefined,
            "display": "flex",
            "height": 55,
            "justifyContent": "center",
            "width": 55,
          }
        }
        tint="dark"
        transitionDuration={300}
      >
        <Image
          backgroundColor="neutral.100"
          borderRadius="l"
          isCached={false}
          loaderType="spinner"
          source={
            Object {
              "uri": "https://pbs.twimg.com/profile_images/1419369145058041856/eshLFaDy_400x400.jpg",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderBottomLeftRadius": undefined,
                "borderBottomRightRadius": undefined,
                "borderRadius": 1000,
                "borderTopLeftRadius": undefined,
                "borderTopRightRadius": undefined,
                "height": "100%",
                "width": "100%",
                "zIndex": 2,
              },
            ]
          }
          tint="dark"
          transitionDuration={300}
        />
      </View>
      <View
        onLayout={[Function]}
        size="s"
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#A2D608",
            "borderRadius": 1000,
            "bottom": -0,
            "display": "flex",
            "height": 15,
            "justifyContent": "center",
            "minWidth": 15,
            "position": "absolute",
            "right": -0,
          }
        }
      />
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "borderRadius": undefined,
        "display": "flex",
        "flexDirection": "row",
        "marginRight": 12,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignSelf": "flex-start",
          "borderRadius": undefined,
          "display": "flex",
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00ffff",
            "borderRadius": 1000,
            "display": "flex",
            "height": 55,
            "justifyContent": "center",
            "width": 55,
          }
        }
      >
        <Text
          accessibilityRole="text"
          accessible={true}
          allowFontScaling={true}
          scaleFontSize={true}
          style={
            Object {
              "color": "#1A2138",
              "display": "flex",
              "fontFamily": "Poppins-Regular",
              "fontSize": 18.4,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 28,
            }
          }
          variant="p1"
        >
          RA
        </Text>
      </View>
      <View
        onLayout={[Function]}
        size="s"
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#FF4B42",
            "borderRadius": 1000,
            "bottom": -0,
            "display": "flex",
            "height": 15,
            "justifyContent": "center",
            "minWidth": 15,
            "position": "absolute",
            "right": -0,
          }
        }
      >
        <Text
          accessibilityRole="text"
          accessible={true}
          allowFontScaling={true}
          scaleFontSize={true}
          style={
            Object {
              "color": "#FFFFFF",
              "display": "flex",
              "fontFamily": "Poppins-Medium",
              "fontSize": 11.5,
              "fontStyle": "normal",
              "fontWeight": "500",
              "lineHeight": 14,
              "paddingHorizontal": 4,
            }
          }
          variant="btn4"
        >
          0
        </Text>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "borderRadius": undefined,
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignSelf": "flex-start",
          "borderRadius": undefined,
          "display": "flex",
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ff0000",
            "borderRadius": 1000,
            "display": "flex",
            "height": 55,
            "justifyContent": "center",
            "width": 55,
          }
        }
      />
      <View
        onLayout={[Function]}
        size="m"
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#6A7BFF",
            "borderRadius": 1000,
            "bottom": -5,
            "display": "flex",
            "height": 25,
            "justifyContent": "center",
            "minWidth": 25,
            "position": "absolute",
            "right": -5,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;
